Modules/_sqlite/module.c

```diff
@@ -66,7 +66,7 @@ static PyObject* module_connect(PyObject* self, PyObject* args, PyObject*
 
     PyObject* result;
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|diOiOip", kwlist,
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|diOiOip", kwlist,
```

Comment:
I don't understand why the function checks arguments.

Why not only relying on Connection constructor to check arguments?

Either remove the whole code to parse arguments, or fix the reference leak: Py_DECREF(database); is needed.

Reply:
> fix the reference leak: Py_DECREF(database); is needed.

Oops, it's not needed, you don't use PyUnicode_FSConverter, sorry. Your code is correct.

Reply:
@Phaqui pointed me on IRC that the function contains a comment explaining that we have to parse all arguments to just extract the "factory" keyword argument... In that case, the code is ok.